### DISCUSSION TITLE
vNext as 4.2 or 4.1.1? vNext soon or in months?

### DISCUSSION THREAD

**Discussion by gerhardol** [gitextensions/gitextensions#11037] [General]
When should next release be?
There are some commits resolved in the milestone (and in release/4.0), a few open (mostly resolved). However, there has not been much fuzz about the issues for some time, maybe this can be skipped. The longer time that passes from 4.1 release, the 'riskier' is a 4.1.1 release: No one is using that specific branch.
The milestone:https://github.com/gitextensions/gitextensions/milestone/68
The remaining issues in the milestone should be handled soon regardless.

There are what I recall no controversial issues in master right now, so 4.2/4.1.1 could be released from master too. There has been some refactoring, there is some noise in the diff.

Another option is to include some more PRs, some open that actually contains features. Personally, I may be less active over the summer here and maybe similar for other contributors from the northern hemisphere. vNext is realisticly earliest in the fall, but then it is close to .NET8.
Regardless, the expected time for including all major changes should be agreed on better than for 4.0 and 4.1. (For 4.0 the release was set before all new features were 'completed' which delayed the release. For 4.1 it was slightly the same, with some changes in 4.0 raising quite some issues just before the planned release, which blocked master for normal dev unnecessarily.)

**Comment by RussKie** [comment]
I'd go for 4.2 having finalised the pull requests actively been worked on.

**Reply by mstv** [reply]
> There are what I recall no controversial issues in master right now, so 4.2/4.1.1 could be released from master too.

👍

> soon or in months? However, there has not been much fuzz about the issues for some time, maybe this can be skipped.

So, I do not have a preference myself.

> Another option is to include some more PRs.

Yes, the bugfix PRs.

The first three graph PRs are not risky IMO. I want them to be merged soon in order to not need to resolve conflicts on all those refactorings and to merge them again and again.
I see no need to integrate them into a sooner 4.2.
The diagonals feature should be tested by a few more early adopters before a release.

> I may be less active over the summer here and maybe similar for other contributors from the northern hemisphere.

Me mainly in July (but much less apples and plums than last year).

**Reply by RussKie** [reply]
My workload is pretty high, but I think we can muster a new release by September (if not earlier).

**Reply by gerhardol** [reply]
If September, the graph changes could be included. But mid August the features to fix for a 4.2 release should be clearly communicated and merge to master be limited.

**Comment by RussKie** [comment]
Hey folks, I think it could be time for us to make a new release (4.2). Is there anything that you feel blocking a release?

**Reply by gerhardol** [reply]
I feel that the PRs for exception handling, grid etc should be merged, then some usage and expected fine tuning.
@mstv?

**Reply by vbjay** [reply]
I think reading all configs system and worktree configs should be taken care of. Not reading system git config is wrong.

**Reply by mstv** [reply]
I do not see a blocking point. There are no(t many) complaints from user neither
- about system and worktree configs (not easy to implement, open discussion) nor
- about the weird behavior of view settings.

I would like to merge #11162 and #11152 yet and do not wait too long with the release.

**Comment by RussKie** [comment]
📢📢📢 I have snaped 4.2 into release/4.0 branch. The master is now targeting 4.3. I've done this to avoid any significant changes (like https://github.com/gitextensions/gitextensions/pull/11240, https://github.com/gitextensions/gitextensions/pull/11242 or https://github.com/gitextensions/gitextensions/pull/10866 accidentally getting merged so close to the release).

Please mark anything that you can't live without in the next release as [4.2](https://github.com/gitextensions/gitextensions/milestone/69).

**Reply by RussKie** [reply]
The RC1 is available here: https://github.com/gitextensions/gitextensions/releases/tag/v4.2-rc1
